tests/cpp/test_low_precision_recipe.cpp

```diff
@@ -253,6 +253,212 @@ TEST_P(MXFP8QuantizationTest, AutoScheduleOp) {
       << ", atol=" << atol << ")";
 }
 
+TEST_F(MXFP8QuantizationTest, AutoScheduleOpHandleNoVectorizedInput) {
+  const auto data_hp_dtype = DataType::Float;
+  const int m = 1024;
+  const int n = 1024;
+
+  std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  createMXFP8QuantizationFusion(fusion.get(), data_hp_dtype);
+
+  FusionExecutorCache fec(std::move(fusion));
+
+  // Create input tensor using custom CUDA allocation with misalignment
+  size_t element_size = 4;
+  size_t total_elements = m * n;
+  size_t buffer_size =
+      total_elements * element_size + 32; // Extra bytes for misalignment
+
+  // Allocate GPU memory with extra space
+  void* gpu_ptr;
+  cudaMalloc(&gpu_ptr, buffer_size);
+
+  // Create tensor from GPU memory at offset of 4 bytes (misaligned)
+  void* misaligned_ptr = static_cast<char*>(gpu_ptr) + 4;
+  auto input_tensor = at::from_blob(
+      misaligned_ptr,
+      {m, n},
+      at::TensorOptions()
+          .dtype(data_type_to_aten(data_hp_dtype))
+          .device(at::kCUDA));
```

Comment:
I vaguely remember we discussed about this, but I don't remember the reason for the explicit alloc.

nitpick: I *think* this looks much simpler.

```
  auto t0 = at::randn({1024*1024+1}).index({at::indexing::Slice(1)});
```